simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/parser/arc/ArcTraceReader.java

```diff
@@ -15,31 +15,42 @@
  */
 package com.github.benmanes.caffeine.cache.simulator.parser.arc;
 
+import static com.github.benmanes.caffeine.cache.simulator.policy.Policy.Characteristic.WEIGHTED;
+
 import java.io.IOException;
+import java.util.Set;
 import java.util.stream.LongStream;
+import java.util.stream.Stream;
 
 import com.github.benmanes.caffeine.cache.simulator.parser.TextTraceReader;
-import com.github.benmanes.caffeine.cache.simulator.parser.TraceReader.KeyOnlyTraceReader;
+import com.github.benmanes.caffeine.cache.simulator.policy.AccessEvent;
+import com.github.benmanes.caffeine.cache.simulator.policy.Policy.Characteristic;
+import com.google.common.collect.Sets;
 
 /**
  * A reader for the trace files provided by the authors of the ARC algorithm. See
  * <a href="http://researcher.watson.ibm.com/researcher/view_person_subpage.php?id=4700">traces</a>.
  *
  * @author ben.manes@gmail.com (Ben Manes)
  */
-public final class ArcTraceReader extends TextTraceReader implements KeyOnlyTraceReader {
+public final class ArcTraceReader extends TextTraceReader {
 
   public ArcTraceReader(String filePath) {
     super(filePath);
   }
 
   @Override
-  public LongStream keys() throws IOException {
-    return lines().flatMapToLong(line -> {
+  public Set<Characteristic> characteristics() {
+    return Sets.immutableEnumSet(WEIGHTED);
+  }  
+
+  @Override
+  public Stream<AccessEvent> events() throws IOException {
+    return lines().flatMap(line -> {
       String[] array = line.split(" ", 3);
       long startBlock = Long.parseLong(array[0]);
       int sequence = Integer.parseInt(array[1]);
-      return LongStream.range(startBlock, startBlock + sequence);
+      return LongStream.range(startBlock, startBlock + sequence).mapToObj(key -> AccessEvent.forKeyAndWeight(key, 512));
```

Comment:
Please keep to 100c line length, so put `.mapToObj` on the next line (similar in other changes, too)